src/core/src/sketch/minhash.rs

```diff
@@ -516,67 +543,123 @@ impl KmerMinHash {
+                    self.seed,
+                    other.max_hash,
+                    self.abunds.is_some(),
+                );
+                //new_mh.add_many_with_abund(&self.abunds)?;
+                return new_mh.compare(other, false)
             }
+        }
+        else {
+            self.check_compatible(other)?;
+
+            if ignore_abundance {
+                if let Ok((common, size)) = self.intersection_size(other) {
+                    Ok(common as f64 / u64::max(1, size) as f64)
+                } else {
+                    Ok(0.0)
+                }
+            } else {
+                if self.abunds.is_none() || other.abunds.is_none() {
+                    // TODO: throw error, we need abundance for this
+                    unimplemented!()
+                }
 
-            let abunds = self.abunds.as_ref().unwrap();
-            let other_abunds = other.abunds.as_ref().unwrap();
-
-            let mut prod = 0;
-            let mut other_iter = other.mins.iter().enumerate();
-            let mut next_hash = other_iter.next();
-            let a_sq: u64 = abunds.iter().map(|a| (a * a)).sum();
-            let b_sq: u64 = other_abunds.iter().map(|a| (a * a)).sum();
-
-            for (i, hash) in self.mins.iter().enumerate() {
-                while let Some((j, k)) = next_hash {
-                    match k.cmp(hash) {
-                        Ordering::Less => next_hash = other_iter.next(),
-                        Ordering::Equal => {
-                            // Calling `get_unchecked` here is safe since
-                            // both `i` and `j` are valid indices
-                            // (`i` and `j` came from valid iterator calls)
-                            unsafe {
-                                prod += abunds.get_unchecked(i) * other_abunds.get_unchecked(j);
+                let abunds = self.abunds.as_ref().unwrap();
+                let other_abunds = other.abunds.as_ref().unwrap();
+
+                let mut prod = 0;
+                let mut other_iter = other.mins.iter().enumerate();
+                let mut next_hash = other_iter.next();
+                let a_sq: u64 = abunds.iter().map(|a| (a * a)).sum();
+                let b_sq: u64 = other_abunds.iter().map(|a| (a * a)).sum();
+
+                for (i, hash) in self.mins.iter().enumerate() {
+                    while let Some((j, k)) = next_hash {
+                        match k.cmp(hash) {
+                            Ordering::Less => next_hash = other_iter.next(),
+                            Ordering::Equal => {
+                                // Calling `get_unchecked` here is safe since
+                                // both `i` and `j` are valid indices
+                                // (`i` and `j` came from valid iterator calls)
+                                unsafe {
+                                    prod += abunds.get_unchecked(i) * other_abunds.get_unchecked(j);
+                                }
```

Comment:
Not related to this PR, but I will run the benchmarks again for this part... the `unsafe` block and `.get_unchecked` are probably unnecessary, the compiler is good at avoiding bound checks and I think it would pick up this case.